### PR TITLE
Support branching in Symbolic Execution

### LINKED ISSUE #5306: Support branching in Symbolic Execution
Add proper branching support in the new Symbolic Execution engine

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #5306

### CODE REVIEW

**Comment by pavel-mikula-sonarsource** [comment]
There's some CI issue, coverage was not computed.

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.cs [hunk 1]
```diff
@@ -61,13 +62,11 @@ public partial class RoslynSymbolicExecutionTest
...
-                "InCatch",
-                "InFinally",
-                "AfterFinally",
-                "Else",
-                "End");
+                "BeforeTry",        // Dequeue for "if" branch
+                "Else",             // Dequeue for "else" branch
+                "InTry",            // Dequeue after "if" branch
+                "End",              // Dequeue after "else" branch, reaching exit block
+                "AfterFinally");    // Dequeue after "if" branch
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.cs:69]: So is the `ToDo` there 😆

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Comment by pavel-mikula-sonarsource** [comment]
There's a known issue with agents that prevents coverage from being processed. The PR was 100% covered in one of the successful runs.

**Pull request merged**